bin/specextract

```diff
@@ -1603,34 +1648,75 @@ def resps(args):
             except OSError as exc:
                 raise IOError(f"Failed to create ARF for {fullfile}") from exc
 
-        try:
-            asphist_temp.close()
-        except NameError:
-            asphist_temp = None
-
-        try:
-            tdetwmap.close()
-        except (NameError,AttributeError):
-            tdetwmap = None
-        
-        try:
-            fef_file.close()
-        except (NameError,AttributeError):
-            fef_file = None
-
-        try:
-            outreg.close()
-        except NameError:
-            outreg = None
-
     if srcbkg == "bkg" and not dobkgresp:
         return None
     else:
```

Comment:
We can remove the `else` line and de-indent the following lines. Also, are we guaranteed that mkresp is one of these three values? If so the last one can remove the if check (maybe changing it to a 

    assert mkresp == "ARF+RMF"

to catch if extra values can be added to `mkresp`. Alternatively the routine could just end in a

    raise ValueError(f"Unknown mkresp={mkresp}")

Reply:
revised in latest commit